### DISCUSSION TITLE
Работа с отчетом (график,функции)

### DISCUSSION THREAD

**Discussion by Murbulat1** [totumonline/totum-ru-issues-and-discussions#384] [Как это сделать?] [answered]
Доброго времени суток, в данный момент изучаю работу с графиками и функциями.  Не понимаю как добавить значение дат которые будут браться из поля в фильтрацию  Т.е. ограничить список фильтрации от и до. 

Создал временную таблицу, и в h_insert ввел следующий код (извините за комментарии, это я для себя их вводил)
![Снимок](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/92213625/2feac7b8-745c-4896-87ae-e7056eaaa4ed)

Как мне в фильтрацию добавить значение дат от и до, которые будут браться из поля?

**Comment by erpic-ru** [comment]
Что-то как-то сложно  :)
Судя по коду, ограничение нужно разместить в переменную list, которая почему-то массив с одним ключом. Можно немного упростить: брать только список и, соответственно, убрать [id] везде. 

`list: selectlist(table: 'spisok pozicii'; field 'id'; where: 'date'>= Поле с датой1; where 'date'<= Поле с датой2;`

Ещё фильтрация (filter) выглядит лишней, учитывая, что уже была выборка по условию (в selectlist).

**Reply by Murbulat1** [reply]
Благодарю, попробую сегодня. На Live Demo , есть "отчет по позициям". Пытаюсь теорию изучить на практике, вот и разбираю данный раздел.  Там есть список товаров,  с позициями.  А график получает из таблицы "отгрузки" , вот думаю как прикрутить по определенным датам , вообщем сложно но интересно :)

**Reply by ttm-alexeypolunin** [reply, marked as answer]
На самом деле все должно быть просто. Вам на старте надо брать selectList указав where две штуки ограничивающие даты.

Если вы хотите в процессе их отфильтровать то берете selectRowList с датами в том числе и listFiler у него параметр key множественный — задаете несколько `key: "date" >= $date_start; key: "date" >= $somthing`
